File: mobile-app/app/src/main/java/com/maroua/pharmaciegarde/data/model/Pharmacy.kt
package com.maroua.pharmaciegarde.data.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Pharmacy(
    val id: Int,
    val name: String,
    val address: String,
    val phone: String,
    @SerializedName("phone_secondary")
    val phoneSecondary: String?,
    val latitude: Double,
    val longitude: Double,
    @SerializedName("image_url")
    val imageUrl: String?,
    val description: String?,
    val city: String,
    val district: String?,
    @SerializedName("has_parking")
    val hasParking: Boolean,
    @SerializedName("has_wheelchair_access")
    val hasWheelchairAccess: Boolean,
    @SerializedName("is_24_hours")
    val is24Hours: Boolean,
    @SerializedName("is_on_duty_today")
    val isOnDutyToday: Boolean,
    @SerializedName("current_schedule")
    val currentSchedule: Schedule?,
    val distance: Double? = null
) : Parcelable

@Parcelize
data class Schedule(
    val id: Int,
    @SerializedName("pharmacy_id")
    val pharmacyId: Int,
    @SerializedName("start_date")
    val startDate: String,
    @SerializedName("end_date")
    val endDate: String,
    @SerializedName("start_time")
    val startTime: String?,
    @SerializedName("end_time")
    val endTime: String?,
    @SerializedName("is_on_duty")
    val isOnDuty: Boolean,
    val notes: String?
) : Parcelable

data class ApiResponse<T>(
    val success: Boolean,
    val data: T,
    val message: String? = null
)

File: backend/app/Models/Pharmacy.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Pharmacy extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'address',
        'phone',
        'phone_secondary',
        'latitude',
        'longitude',
        'image',
        'description',
        'city',
        'district',
        'has_parking',
        'has_wheelchair_access',
        'is_24_hours',
        'is_active',
    ];

    protected $casts = [
        'latitude' => 'decimal:8',
        'longitude' => 'decimal:8',
        'has_parking' => 'boolean',
        'has_wheelchair_access' => 'boolean',
        'is_24_hours' => 'boolean',
        'is_active' => 'boolean',
    ];

    protected $appends = ['image_url'];

    /**
     * Accesseur pour l'URL complète de l'image
     */
    public function getImageUrlAttribute(): ?string
    {
        if ($this->image) {
            return asset('storage/' . $this->image);
        }
        // Image par défaut
        return asset('images/pharmacy-default.png');
    }

    public function schedules(): HasMany
    {
        return $this->hasMany(Schedule::class);
    }

    public function ratings(): HasMany
    {
        return $this->hasMany(Rating::class);
    }

    public function currentSchedule()
    {
        return $this->schedules()
            ->where('start_date', '<=', now())
            ->where('end_date', '>=', now())
            ->where('is_on_duty', true)
            ->first();
    }

    public function isOnDutyToday(): bool
    {
        return $this->currentSchedule() !== null;
    }

    public function averageRating(): float
    {
        return round($this->ratings()->avg('rating') ?? 0, 1);
    }

    public function ratingsCount(): int
    {
        return $this->ratings()->count();
    }
}
